osquery/sql/virtual_sqlite_table.cpp

```diff
@@ -112,4 +114,25 @@ Status genTableRowsForSqliteTable(const fs::path& sqlite_db,
 
   return Status{};
 }
+
+Status getSqliteJournalMode(const fs::path& sqlite_db) {
+  TableRows result;
+  auto status = genTableRowsForSqliteTable(
+      sqlite_db, "PRAGMA journal_mode;", result, true);
+  if (!status.ok()) {
+    return status;
+  }
+  if (result.size() < 1) {
+    VLOG(1) << "PRAGMA query returned empty results";
+    return Status(1, "Could not retrive journal mode");
```

Comment:
Should we distinguish between these two errors?

Reply:
Those checks are there because I'm paranoid about everything, but in the real world if the query does not fail (and the method returns early) it will have some results (unless libsqlite is particularly broken), so I don't think it's worth having different statuses (plus ATM the status error message is not read on the receiving end of the function).